rpc/v9/subscriptions.go

```diff
@@ -517,6 +542,19 @@ func (h *Handler) SubscribeTransactionStatus(ctx context.Context, txHash *felt.F
 			}
 			return nil
 		},
+		onPreLatest: func(
+			ctx context.Context,
+			id string,
+			sub *subscription,
+			preLatest *core.PreLatest,
+		) error {
+			if lastStatus < TxnStatusAcceptedOnL2 {
+				if lastStatus, err = h.checkTxStatus(ctx, sub, id, txHash, lastStatus); err != nil {
+					return err
+				}
+			}
+			return nil
```

Comment:
nit: Is it possible to create a function for this? I think it's the same as `onPendingData`